### PR TITLE
[deliver] update deliver/runner.rb to handle both ipa and pkg paths.

### PR DESCRIPTION
**Pull request by snatchev**
See [#19659 ](https://github.com/fastlane/fastlane/issues/19659#issuecomment-1063018478) for rationale.

### CODE REVIEW

File: deliver/lib/deliver/runner.rb [hunk 1]
```diff
@@ -162,28 +162,25 @@ def upload_binary
...
           package_path: "/tmp",
           platform: options[:platform]
         )
+      else
+        UI.user_error!("no suitable file found for upload for platform: #{options[:platform]}")
       end
+        
 
       transporter = transporter_for_selected_team
       result = transporter.upload(package_path: package_path, asset_path: upload_ipa || upload_pkg)
```
Comment (krzysiek84) [inline deliver/lib/deliver/runner.rb:186]: Hey @snatchev  in this line the asset_path might still be wrong. For clarity i would duplicate this line in both 'switch' cases

Comment (krzysiek84) [reply deliver/lib/deliver/runner.rb:186]: Something like this
```
 platform = options[:platform]
      transporter = transporter_for_selected_team
      
      case platform
        when "ios", "appletvos"
          package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(
            app_id: Deliver.cache[:app].id,
            ipa_path: options[:ipa],
            package_path: "/tmp",
            platform: platform
          )
          result = transporter.upload(package_path: package_path, asset_path: upload_ipa)

        when "osx"
          package_path = FastlaneCore::PkgUploadPackageBuilder.new.generate(
            app_id: Deliver.cache[:app].id,
            pkg_path: options[:pkg],
            package_path: "/tmp",
            platform: platform
          )
          result = transporter.upload(package_path: package_path, asset_path: upload_pkg)

        else
          UI.user_error!("no suitable file found for upload for platform: #{options[:platform]}")
        end

```

Comment (snatchev) [reply deliver/lib/deliver/runner.rb:186]: @krzysiek84 this looks good to me, feel free to cook up a real PR for this change, and ping me for a review.

**Review by krzysiek84 — COMMENTED** [review]
Looks good to me 👍

**Review by crazymanish — APPROVED** [review]
this looks good! 🚀

**Comment by joshdholtz** [comment]
@snatchev Added a commit to update that `verify_binary` method to use the same logic!

**Comment by snatchev** [comment]
@joshdholtz great! Thanks so much!

**Pull request merged**